### DISCUSSION TITLE
Roadmap

### DISCUSSION THREAD

**Discussion by opekope2** [opekope2/StaffMod#31] [General]
# Staff Mod development roadmap (in no particular order)

## Items in a staff

* [ ] Amethyst block
* [x] Anvil, chipped anvil, damaged anvil
* [ ] Bedrock
* [x] Bell
* [x] Bone block
* [ ] Brewing stand
* [x] Cake
* [x] Campfire, soul campfire
* [ ] Cauldron
* [ ] Command block
* [ ] Crafting table
* [x] Diamond block
* [ ] Dispenser
* [ ] Dripstone block
* [x] Emerald block
* [ ] Enchanting table
* [ ] End portal frame
* [x] Furnace, blast furnace, smoker
* [x] Gold block
* [x] Lightning rod
* [x] Magma block
* [x] Netherite block
* [ ] Note block
* [ ] Obsidian
* [ ] Redstone block
* [ ] Sculk Sensor
* [x] Snow block
* [ ] Spawner
* [x] TNT
* [x] Wither skeleton head
* [x] Wool

### Not yet decided

* [ ] Minecraft icon
* [ ] Barrier
* [ ] Bee nest
* [ ] Bookshelf
* [ ] Cobblestone, granite, quartz, and other stone variants
* [ ] Cobweb
* [ ] Grindstone
* [ ] Ice
* [ ] Iron block
* [ ] Logs
* [ ] Lapis lazuli block
* [ ] Lucky block
* [ ] Netherrack
* [ ] Piston
* [ ] Red/brown mushroom block
* [ ] Blocks not showcased in the video

### Not yet decided

* [ ] Animations

**Comment by thr3zims** [comment]
Is the lightning rod staff supposed to have such short range? I can't use it in survival without hitting myself.

**Reply by opekope2** [reply]
I haven't changed its reach, but I do plan to increase it, I don't yet know how much

**Reply by FuryJack07** [reply]
I'd say 15 extra blocks of radius if you want it to be powerful, otherwise 5 extra blocks should be enough to use it without hitting yourself

**Reply by clownfish7050** [reply]
Also the campfire sometimes pushed me back. Is this intentional?

**Reply by opekope2** [reply]
@FuryJack07 15 extra blocks is an entire chunk, but I don't mind extending the reach by a few extra blocks

**Reply by opekope2** [reply]
@clownfish7050 Yes, you can fly with it. [The wiki describes how to trigger it](https://opekope2.github.io/StaffMod/staff.html#campfire-soul-campfire).

**Reply by opekope2** [reply]
I'll fix the range in the first 1.20.5 release (the reach attribute arrived in 23w51a)

**Reply by Devbuilder10011** [reply]
You think you can add something so when you shift right click you can grab placed blocks with the staff?

**Reply by opekope2** [reply]
[You can pick up blocks from the world in the latest version](https://opekope2.dev/StaffMod/staff.html#no-item).

**Comment by FuryJack07** [comment]
How are you planning to implement unobtainable blocks like the Command block, end-frame and mob spawner?

I know that they're in phase 2, just curios

**Reply by opekope2** [reply]
Configurable staffs are not yet in the code. Brewing stand, dispenser, crafting table are some examples of this. The plan is to configure them with sneak+right click, and use them with right click (some will also have left click functionality)

Command block: configurable staff. It's not fast to change it, but if you have a better idea, LMK.
Mob spawner: configurable staff.
End portal frame: I'll use Immersive Portals if present. If not, end portal frame staff will not be usable. Left click to place one portal, right click to place the other one (just like in Portal).

**Reply by opekope2** [reply]
WIP command block staff. It's not ready, don't expect it in the next release.
This implementation also spams the server console, which admins don't want. Admins also don't want users executing arbitrary commands, so I see no point adding this on a survival server. It may be interesting to goof around in single player, but idk if I should let survival players use it (vanilla Minecraft doesn't allow the use of command blocks in survival, but AvM universe does).

![Log spam](https://github.com/opekope2/StaffMod/assets/20487556/58c128c4-f342-4cac-92ec-746b3070b399)
![Config screen](https://github.com/opekope2/StaffMod/assets/20487556/b1b83b04-58f2-466b-b5d3-6fb8ab41d548)

**Reply by ScoutTheSaneless** [reply]
mmmm yes, ultimate power.....muahahahahahahahaHAHAHAHAHAHAHA!!!
_ahem_
honestly, the way i see it, if the player is given a way to access command blocks in survival, be it a crafting recipe, a loot table, a villager/trader trade, or a structure that adds one, they should be able to use it, otherwise, tough cookies

**Comment by FuryJack07** [comment]
Idea: Hold shift right-click on a usable block to absorb it. The staff is supposed to be an end-game item, right? then make it so that the time needed is equal to using a diamond tool plus 1 second so that insta-mine blocks still take a second to mine.

If you can, make it affected by mining fatigue.

**Reply by FuryJack07** [reply]
this way you can also get Spawners and end portal frames

**Reply by opekope2** [reply]
Makes sense, King Orange does this. I plan to implement it in the future in some capacity.
However, this can be easily exploited to do things Mojang didn't intend to allow, like removing unbreakable blocks, moving end portal home, and obtaining unobtainable blocks if I don't put some limitations in place + unbreakable blocks will take forever the way you described. I don't want to add a way to obtain the unobtainable, but if you already have them (Minecraft betas, 1.12, or any other version), using them will be allowed

**Reply by ScoutTheSaneless** [reply]
or other methods like mods (i.e. Apotheosis for spawners), datapack/CraftTweaker/KubeJS custom recipes/loot tables/trades for literally any of them. and besides, we Java users can use bedrock breakers to carve holes through to the void and above the nether and Mojang havent done anything to completely wipe it.

**Comment by TheDogeGuy** [comment]
i love this mod
i really love this mod, all i think it needs besides pratically every block in the game, is to like shift right click blocks besides some blocks like bedrock and end portal frames, and then just putting it into the staff, as king orange does do this

**Reply by opekope2** [reply]
Thanks 😁
I'll let players add blocks to the staff directly from the world after I find a balanced way to do so.

**Comment by witheredpathogen** [comment]
my ideia is adding a "mode" to the command block staff which after you press Z, you can select abilities by using 1 2 3 4 etc on your keyboard, and make those abilities based on king orange's abilities, ability 1 like a barrier/bedrock/obsidian shield, ability 2 like a laser that deals a ton of damage, ability 3 which creates blocks in a configurable manner, like 1x1, 2x2, 3x3x3 and you can choose any block to replace that, the reverse with ability 4.

**Reply by witheredpathogen** [reply]
also, is there a discord server i could potentially join? i wanna see the progress in this mod because it looks pretty cool/is pretty cool.

**Reply by opekope2** [reply]
There's no discord server because this is a project I do for fun, and I don't want to deal with moderation. I'm not looking for drama, which, let's be real, finds every community.

**Reply by witheredpathogen** [reply]
Not big ones, also what did you think of my ideia? i though of a few more like: you can choose whatever block u wanna use in the command block staff for shields etc everything block related with G, which pops up a little GUI which displays the block in 3d rotating and then you type the block name and tab for it to autocomplete to the one it thinks ur doing, same with mobs but its Y. As for potion effects, just make it like a ctrl + scroll where u scroll through the effects, it wont affect ur hotbar tho.

**Reply by BoB10011** [reply]
That sounds cool but to much work for what I'm guessing is the 1 guy

**Reply by opekope2** [reply]
Ability 2 is the redstone block staff, the others sound like the Game Icon (which is probably as much work as the mod itself).

**Reply by mritaliano565** [reply]
perhaps you could make someone elses game icon mod compatible for this kinda thing (not sure if it'll help or not just putting it out there)

**Comment by KillerBean208** [comment]
Hello @opekope2. Say, will you consider opening up a Discord server regarding this mod for sneak peeks and updates?

**Reply by opekope2** [reply]
I don't have the time and motivation to moderate a Discord server.
For updates, you can subscribe on Modrinth and/or CurseForge. Sometimes I upload a video to YouTube.
Sneak peeks? I usually develop an update in about a couple of days to a couple of weeks (depending on the block functionality), then I release it when it's finished, and do other things for another couple of weeks, so there are usually not a lot of sneak-peek-worthy things.

**Comment by NAILTOP** [comment]
maybe it’s worth making it so that when indestructible blocks are placed from the world, they do not disappear from the world itself, but appear in the staff, and when they are removed from the staff, they disappear completely.

**Reply by NAILTOP** [reply]
and a couple more ideas for unsolved blocks:
barrier (allows you to place and destroy barrier blocks, which can be highlighted with a hotkey)
bookshelf (throws books and paper that deal little damage and knocks back a little)
web (can place web blocks or can work as a grappling hook, like from the recent April Fools' update)
ice (allows you to slow down mobs, turn water into ice and just place ice)
iron block (same as other gem blocks, only on a smaller scale)
piston (can strongly push mobs away, and also allows you to bounce off blocks)
lucky block (when used, turns into a random block from available ones)

**Reply by opekope2** [reply]
Not removing indestructible blocks from the world and making them disappear from the staff when removed is a great idea

**Comment by Console-Name** [comment]
I cant find out how to put the block in the staff. I tried everything from offhand to throwing them together to right clicking the placed block! Help pls

**Reply by opekope2** [reply]
https://opekope2.dev/StaffMod/items.html#insert-an-item-into-the-staff

**Reply by Console-Name** [reply]
Bro thanksss

**Comment by charmander9695** [comment]
I'm going to say this, king orange's staff is a lot over powered

**Reply by opekope2** [reply]
I will remove infinite block (wool, carpet) placement and projectile shooting next update with a new game rule to enable it.
Do you have any suggestions for balancing?

**Reply by TeboMC** [reply]
How will the wool staff work without the infinite wool?

**Reply by opekope2** [reply]
Just like bow and crossbow: it will take the wool out of your inventory

**Reply by TeboMC** [reply]
Oh, alright, seems like a good plan! If it does that, can you allow the staff to build on top of wool that the staff makes?

**Reply by charmander9695** [reply]
making the camp fire burn out after long uses is an idea i have

**Reply by mritaliano565** [reply]
pls make it so it can also automatically take ammo out of shulker boxes, it would be very useful especially since a lot of them are extremely rapid fire (and also easier to have multiple staffs at once so you can have a lot of ammo for each one)

**Comment by avaandavmmodmaker** [comment]
I'm making an ava mod where you can become a stickman oc and one of the charaters or a human just letting you know as I think you're an expert and your mod inspired me to make it (and another mod inspired me to make it too) but yours mainly inspired me keep up the good work you legend of a mod dev

**Reply by avaandavmmodmaker** [reply]
also you could make it so you cant get the creative only blocks in survival mode (also a idea for the game icon it could turn you into creative mode when you hold it and you could make it so you can only get it with commands)

**Reply by charmander9695** [reply]
what is going to be the name

**Reply by avaandavmmodmaker** [reply]
> what is going to be the name

its called ava mod right now
also its out on curseforge right now

**Reply by mritaliano565** [reply]
is it on modrinth or will be on modrinth?

**Comment by KillerBean208** [comment]
When is the Lapis block ready for the staff btw?

**Reply by KillerBean208** [reply]
@opekope2?

**Reply by TeboMC** [reply]
I dont think he really wants to reply, or he hasnt seen you,re message yet, please do not annoy him.

**Reply by KillerBean208** [reply]
Easy for you to say. Why wouldn’t he want to answer that question tho?

**Reply by opekope2** [reply]
@TenebrisD90 Chill out bro :skull:
I haven't decided it yet.
I have other priorities in the first half of 2025 I'll add other blocks first.

**Reply by KillerBean208** [reply]
Oh... I see. Well regardless, take your time

**Reply by charmander9695** [reply]
take all the time you need, making and balancing this mod will take a lot of time to do

**Reply by mritaliano565** [reply]
its hilarious that a promotion for another mod got called out for the second time on this page and both times thedogeguy was the one calling them out

**Comment by King-Orange123** [comment]
Hey, will you ever release or have a plan to release the minecraft icon after coding all the blocks?

**Comment by TeboMC** [comment]
When will AvM Staff Mod Updates/Support Happen/Continue? Your Modrinth Page Is A Bit Misleading You Know? Not Tryna Be Mean, Just Asking/Saying.

**Reply by KillerBean208** [reply]
I think the author might have abandoned this mod. It’s been nearly a year since its last update

**Reply by TeboMC** [reply]
.. oh :( i will miss this mod :(

**Reply by opekope2** [reply]
I wasn't modding much in Q1 and Q2 because of my studies. In July, I *did return* to modding.
I'm currently developing OptiGUI, because
* it has much larger user base
* AvM Staff Mod has received more attention than OptiGUI for the entirety of 2024
* With the next update, it will reach feature completion

I have finite free time. Sometimes I have burnout from modding, but I still want to continue AvM Staff Mod one day.

**Reply by TeboMC** [reply]
> I wasn't modding much in Q1 and Q2 because of my studies. In July, I _did return_ to modding. I'm currently developing OptiGUI, because
> 
> * it has much larger user base
> * AvM Staff Mod has received more attention than OptiGUI for the entirety of 2024
> * With the next update, it will reach feature completion
> 
> I have finite free time. Sometimes I have burnout from modding, but I still want to continue AvM Staff Mod one day.

Nice, the staff mod will continue :D But when would it continue? 2026? this year?

**Reply by opekope2** [reply]
If I delay the command block staff, this year (which I hear, would be preferred over command blocks).

**Reply by KillerBean208** [reply]
Well, I hope we get some thing this year or the next

**Reply by KillerBean208** [reply]
You have no idea how messed up this year has been

**Reply by TeboMC** [reply]
hi tenebris, do u have discrod? can i add u on it by chance?
also to opekope, please delay comamnd block staff for a later release, we need more updates, the hard staffs/complex ones can be later.

**Comment by TeboMC** [comment]
Hey OpeKope2, when would the next AvM Staff Update Drop? And what features would be inside of it? Would it be this year? Before 2026? Christmas? Atleast give us a time, not a exact time, but atleast a time.

**Reply by opekope2** [reply]
1. https://github.com/opekope2/StaffMod/discussions/31#discussioncomment-14592597
2. Lapis staff, bug fixes, recipe changes mostly
3. https://github.com/opekope2/StaffMod/discussions/31#discussioncomment-14592597
4. https://github.com/opekope2/StaffMod/discussions/31#discussioncomment-14592597
5. https://github.com/opekope2/StaffMod/discussions/31#discussioncomment-14592597

**Reply by TeboMC** [reply]
alright. Looking forward to it!

**Reply by Pandzio21** [reply]
yo @opekope2 i added the lapis staff to your mod pretty easy i just added the items just a duplicate of the king staff and boom works i even made the command block staff but  for now it only makes lightning bolts with no cooldown i will try to make the idea of the abilities to choose from 1 to 4 if you want i can help you with the mod just say i too make mods but didnt post any here on github
<img width="491" height="505" alt="image" src="https://github.com/user-attachments/assets/03b2a93a-9eb7-461c-9ce9-28b3aaf41b31" />

<img width="992" height="794" alt="image" src="https://github.com/user-attachments/assets/1220bc94-8efb-4b0d-81de-1d063e5d9bf8" />

**Reply by mritaliano565** [reply]
is pandzio21 helping opekope2 with the mod or is pandzio21 just offering help?

**Reply by Pandzio21** [reply]
Offering

**Reply by opekope2** [reply]
I opened a discussion about the blue staff, I need feedback on the obtaining process: #74

**Comment by TeboMC** [comment]
Question: are you delaying the command block staff? (So the mod could hopefully get updated this year? Because i remember some things, you saying command block staff somtimes in january or somthing, thats what i remember)

**Comment by TeboMC** [comment]
Will the update be in.. October..? November.. December?!

**Reply by KillerBean208** [reply]
Give him time

**Reply by TeboMC** [reply]
> Give him time

i am impatient..

**Reply by mritaliano565** [reply]
i get people can get imatient especially when its a very good mod, but sometimes not even developers know when they'll finish

**Comment by mritaliano565** [comment]
so hows the skulk sensor gonna work, is it gonna throw the player to the side when theres a projectile or something like in AvM?

**Reply by opekope2** [reply]
> is it gonna throw the player to the side when theres a projectile

That's basically how I imagine it

**Reply by TeboMC** [reply]
I am excited for the next staff mod update! :D It must be getting close (i hope) because it is nearly the end of this year, so it might happen in december.

**Comment by mritaliano565** [comment]
i have idea for logs and stones, maybe you could right click on a flat area and select 1 of 5 pre-made things of the variant (like for log it could make a simple wooden house, or a row of stairs to go up, etc, or stone could make a statue, or a blacksmith)

**Comment by TeboMC** [comment]
Hello. We need this mod updated. This mod is super good, all the fans need it. - From All Of The Alan Becker Fans In The World

**Reply by Pandzio21** [reply]
if you want the blue staff i can give you the mine modified version with it

**Comment by Pandzio21** [comment]
yo guys 
<img width="1511" height="939" alt="image" src="https://github.com/user-attachments/assets/9311b45f-0777-42d7-a4fa-855bffee0a63" />
rate this

**Reply by TeboMC** [reply]
HOW DID YOU get that? tell me how plz

**Reply by Pandzio21** [reply]
i made it myself i already said im a developer or scripter whatever you want to call it

**Reply by TeboMC** [reply]
Sry i didnt see t he other msg, im super excited, my discord is tebomc i rly wanna have commadn block staff :D also ur a nice coder too, u and opekope could team up

**Reply by Pandzio21** [reply]
i would like that very much i actually have all the powers

**Reply by opekope2** [reply]
We can talk about collab, you can find me discord (same username). I may take some time to respond tho.

**Reply by mritaliano565** [reply]
10/10

**Reply by FunkDragon1233** [reply]
Hey Pandzio is your mod playable for the public? cuz that is rlly cool

**Comment by Pandzio21** [comment]
My command block staff showcase please @opekope2 see this its a concept
https://streamable.com/s1l5rt 
sorry for link but file was too big

things I added
<img width="892" height="435" alt="image" src="https://github.com/user-attachments/assets/34139323-1d0b-4206-9beb-d7544ccb683a" />
grab beam

<img width="701" height="567" alt="image" src="https://github.com/user-attachments/assets/a07494e3-63b2-447f-8ee1-8a036aabfa6d" />

electrify blocks 

<img width="743" height="581" alt="image" src="https://github.com/user-attachments/assets/ed4789ce-880c-4d75-80f5-3ca40952aa72" />

easy blocks place (block choosing not done yet)

<img width="200" height="446" alt="image" src="https://github.com/user-attachments/assets/a841f3f5-76d3-4d86-bae8-d5988b0272a2" />

flight by staff (animation not done yet)

**Reply by opekope2** [reply]
This is really cool

**Reply by mritaliano565** [reply]
holy moly howd ya manage that? this is awesome! though maybe make less particles for the grab beam. also how would you un-electrify a block?

**Reply by mritaliano565** [reply]
also howd you make it so fast? not long ago you said that you just made the blue staff with command block but only could do lightning strikes, a couple days later and its already almost an exact replica of the one in AvM

**Reply by Pandzio21** [reply]
idk i was bored

**Reply by Pandzio21** [reply]
> holy moly howd ya manage that? this is awesome! though maybe make less particles for the grab beam. also how would you un-electrify a block?

btw do you like the choose block gui? 
<img width="1289" height="695" alt="image" src="https://github.com/user-attachments/assets/4f564eb6-5f76-4a1f-9dfa-c4ccf92eb80c" />

**Reply by mritaliano565** [reply]
oh wow thats awesome and very handy, at first i thought the staff with command block would take a long time but at this rate it'll be in the next update!

**Reply by Pandzio21** [reply]
Im not working with @opekope2 yet idk he doesn't answers on discord

**Reply by mritaliano565** [reply]
oh ok, so how did you get skilled at this stuf anyways, did you just do it randomly in your spare time and eventually got good, or something else?

**Reply by Pandzio21** [reply]
By time i got better

**Comment by mritaliano565** [comment]
the thing is when i see maybe animations on phase 3 i try to think of possible new animations, but i cant, the animations already used are exactly how id imagine steve if he actually gained the staff

**Comment by TeboMC** [comment]
Any news on the Staff Mod?

**Reply by opekope2** [reply]
Be patient!
I announced the update for December.
It's November.

**Reply by TeboMC** [reply]
Ohh, im super excited for ur mod, hopefully its still good at this update, any new staff blocks tho?

**Comment by TeboMC** [comment]
Could you give us a few hints of what other staffs ur adding? like the emerald block staff and etc?

**Comment by TeboMC** [comment]
Hows the mod going?

**Reply by opekope2** [reply]
Almost update time, I just need to test it a bit more, and write a change log.

**Comment by KillerBean208** [comment]
It’s been a year since its last update. I hope it’s going well

**Reply by opekope2** [reply]
Just pushed the new code to `dev`

**Comment by KillerBean208** [comment]
It’s almost Christmas, no?

**Reply by opekope2** [reply]
Yes. Hopefully I can release the update tomorrow.

**Reply by TeboMC** [reply]
> Yes. Hopefully I can release the update tomorrow.

Yay

**Comment by TeboMC** [comment]
hi @opekope2 , do you know the next date a new staff mod version will drop? Anytime soon? Is there a particullar time? Will it be later? Soon? Never? Because, i really like this AvM Staff Mod with all the staffs, and i have found a mod that adds ALL THE STAFF BLOCKS already, you have competition. I really am looking forward for your new staff update. I wonder what other staffs there will be as well.
this is the mod that adds all of the blocks, you could get ideas from it:
https://modrinth.com/mod/staff-of-the-king-orange
